### PR TITLE
Feature/tech 446 only lint changed projects

### PR DESCRIPTION
**Pull request by Jorg88**
- Also update black commands

----
### 📕 [TECH-446](https://vandebron.atlassian.net/browse/TECH-446) Linting should be done for changed projects jorgpost@vandebron.nl 
Currenlty the master pipeline runs `mpyl projects lint` and the monorepo pipelines used to run it as well. It however blocked people developing when someone commits a broken yaml.

We should lint the changed projects in the monorepo pipeline and all projects in the master pipeline (no change required there)

🏗️ Build [4](https://jenkins.k8s-serv-backend.vdbinfra.nl/job/MPyL%20Pipeline%20-%20Test/job/PR-159/4/display/redirect) ✅ Successful, started by _Jorg Post_  
🚀 *[cloudfront-service](https://cloudfront-service-159.test.nl/)*, *job*, *kong-sync*, *[nodeservice](https://nodeservice-159.test.nl/)*, *[sbtservice](https://sbtservice-159.test.nl/)*, *sparkJob*  

[TECH-446]: https://vandebron.atlassian.net/browse/TECH-446?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/mpyl/cli/projects.py [hunk 1]
```diff
@@ -99,24 +100,45 @@ def show_project(obj, name):
...
+        changes = (
+            obj.cli.repo.changes_in_branch_including_local()
+            if branch
+            else obj.cli.repo.changes_in_merge_commit()
+        )
+        build_set = find_build_set(obj.cli.repo, changes, False)
+        for all_projects in build_set.values():
+            for project in all_projects:
+                if project.path not in project_paths:
+                    project_paths.append(project.path)
```
Comment (koeves) [inline src/mpyl/cli/projects.py:126]: you can use a set for this

File: src/mpyl/cli/projects.py [hunk 2]
```diff
@@ -99,24 +100,45 @@ def show_project(obj, name):
     )
 
 
-@projects.command(help="Validate the yaml of found projects against their schema")
+@projects.command(help="Validate the yaml of changed projects against their schema")
+@click.option(
+    "--all",
+    "all_",
```
Comment (koeves) [inline src/mpyl/cli/projects.py:106]: Why do we need the underscore?

File: src/mpyl/cli/projects.py [hunk 3]
```diff
@@ -99,24 +100,45 @@ def show_project(obj, name):
...
+    is_flag=True,
+    help="Validate all project yaml's, regardless of changes on branch",
```
Comment (koeves) [inline src/mpyl/cli/projects.py:108]: Do we need to add default=False, or is that implied already?

**Review by koeves — APPROVED** [review]
Looks good! Just small things

File: src/mpyl/cli/projects.py (hunk 2 above)
Comment (Jorg88) [reply src/mpyl/cli/projects.py:106]: because `all` is a reserved name in python apparently

File: src/mpyl/cli/projects.py (hunk 3 above)
Comment (Jorg88) [reply src/mpyl/cli/projects.py:108]: I copied it from the `build run` cmd and there we don't have a default?

Comment (koeves) [reply src/mpyl/cli/projects.py:108]: Should be fine then

**Pull request merged**